src/System.IO/tests/BinaryWriter/BinaryWriterTests.cs

```diff
@@ -0,0 +1,391 @@
+            dr2.Dispose();
+            mstr.Dispose();
+        }
+
+        [TestMethod]
+        public static void BinaryWriter_CtorAndWriteTests1_Negative()
+        {
+            BinaryWriter dw2;
+            MemoryStream mstr;
+
+            // [] Should throw ArgumentNullException for null argument
+            Assert.Throws<ArgumentNullException>(() => { dw2 = new BinaryWriter(null); }, "Error_1099x! Expected exception not thrown");
+
+            // [] Can't construct a BinaryWriter on a readonly stream
+            mstr = new MemoryStream(new byte[10], false);
+            Assert.Throws<ArgumentException>(() => { dw2 = new BinaryWriter(mstr); }, "Error_98t4y! Expected exception not thrown");
+            mstr.Dispose();
+
+            // [] Can't construct a BinaryWriter with a closed stream
+            mstr = new MemoryStream();
+            mstr.Dispose();
+            Assert.Throws<ArgumentException>(() => { dw2 = new BinaryWriter(mstr); }, "Error_98t4y! Expected exception not thrown");
+        }
+
+        [TestMethod]
+        public static void BinaryWriter_EncodingCtorAndWriteTests()
+        {
+            MemoryStream ms2;
+            BinaryReader sr2;
+            BinaryWriter sw2;
+            String str2;
+
+            // [] Test that we can read and write UTF8
+            ms2 = new MemoryStream();
+            sw2 = new BinaryWriter(ms2, Encoding.UTF8);
+            sw2.Write("This is UTF8\u00FF");
+            sw2.Flush();
+            ms2.Position = 0;
+            sr2 = new BinaryReader(ms2, Encoding.UTF8);
+            str2 = sr2.ReadString();
+            if (!str2.Equals("This is UTF8\u00FF"))
+            {
+                Assert.Fail("Error_1y8xx! Incorrect string on stream");
+            }
+            sr2.Dispose();
+            ms2.Dispose();
+
+            // [] Test that we can read and write BigEndian
+            string testString = "This is BigEndianUnicode\u00FF";
+            ms2 = new MemoryStream();
+            sw2 = new BinaryWriter(ms2, Encoding.BigEndianUnicode);
+            sw2.Write(testString);
+            sw2.Flush();
+            ms2.Position = 0;
+            sr2 = new BinaryReader(ms2, Encoding.BigEndianUnicode);
+            str2 = sr2.ReadString();
+            Assert.AreEqual(testString, str2, "Error_8f7yv! Incorrect string on stream==" + str2);
+            sr2.Dispose();
+            ms2.Dispose();
+
+            // [] Test that we can read and write Unicode
+            testString = "This is Unicode\u00FF";
+            ms2 = new MemoryStream();
+            sw2 = new BinaryWriter(ms2, Encoding.Unicode);
+            sw2.Write(testString);
+            sw2.Flush();
+            ms2.Position = 0;
+            sr2 = new BinaryReader(ms2, Encoding.Unicode);
+            str2 = sr2.ReadString();
+            Assert.AreEqual(testString, str2, "Error_f897h! Incorrect string on stream==" + str2);
+            sr2.Dispose();
+            ms2.Dispose();
```

Comment:
I can't make this a theory because the Encoding.UTF8, etc, properties are not constant expressions, and I cannot use them as arguments to an attribute.  I just did factor them though.